context.go

```diff
@@ -258,6 +274,43 @@ func getAvailableContexts() ([]os.FileInfo, error) {
 	return files, nil
 }
 
+func updateContextFile(file map[string]string, key string, value string) (map[string]string, error) {
+	contextValues := map[string]string{}
+	var validContextKey bool
+
+	for k, v := range file {
+		contextValues[k] = v
+		if k == key {
+			validContextKey = true
+			if v == value {
+				return nil, fmt.Errorf("value: %v already set for %v \n", v, k)
+			}
+			if k == config.EnvFnAPIURL {
+				err := ValidateAPIURL(value)
+				if err != nil {
+					return nil, err
+				}
+			}
+			contextValues[k] = value
+		}
+	}
+
+	if !validContextKey {
+		contextValues[key] = value
+	}
+
+	fmt.Printf("Successfully updated %v: %v \n", key, value)
+	return contextValues, nil
+}
+
+func ValidateAPIURL(apiURL string) error {
+	_, err := url.ParseRequestURI(apiURL)
```

Comment:
I'm not sure if that is a valid method to validate the Fn API URL endpoint,  a good starting point is: https://github.com/fnproject/cli/blob/master/client/api.go#L30 see what we check there and try to make a validation rule out of that logic.